src/core/packages/saved-objects/migration-server-internal/src/model/create_batches.ts

```diff
@@ -131,7 +131,13 @@ export function createBatches({
 
   // create index (update) operations for all transformed documents
   for (const document of documents) {
-    const bulkIndexOperationBody = createBulkIndexOperationTuple(document, typeIndexMap);
+    const idChanged = document._source.originId && document._id !== document._source.originId;
+    const bulkIndexOperationBody = createBulkIndexOperationTuple(
+      document,
+      typeIndexMap,
+      // if the id changed, we shouldn't use optimistic concurrency control since it will fail when writting
+      !idChanged
+    );
```

Comment:
I would recommend moving this logic directly into `createBulkIndexOperationTuple`, where it truly belongs, and keep this code untouched.

While debugging, it's often tricky to trace back and keep track of where all the parameters originated.
Then there will be just one extra unit test covering the changed ID. The solution will be self-explanatory and will not require a comment (zero comments is always better 😄 ).

It will also resolve the CI metrics notice about public APIs missing comments.